app/controllers/api/public/v1/courses_controller.rb

```diff
@@ -3,23 +3,21 @@ module Public
     module V1
       class CoursesController < API::Public::V1::ApplicationController
         def index
-          render json: {
-            data: [
-              {
-                id: "124",
-                type: "Course",
-                attributes: {
-                  code: "3GTY",
-                  provider_code: "066",
-                  age_minimum: 10,
-                  age_maximum: 14,
-                },
-              },
-            ],
-            jsonapi: {
-              version: "1.0",
-            },
-          }
+          render jsonapi: paginate(courses), include: include_param, class: API::Public::V1::SerializerService.call
+        end
+
+      private
+
+        def courses
+          @courses ||= recruitment_cycle.courses
+        end
+
+        def recruitment_cycle
+          @recruitment_cycle ||= RecruitmentCycle.find_by!(year: params[:recruitment_cycle_year])
```

Comment:
I just realised something ... we have two routes that get all the courses for a recruitment cycle in the v3 API:

```
/api/v3/recruitment_cycles/:recruitment_year/courses
/api/v3/courses
```

The later one defaults to the current recruitment year and is what Find uses to retrieve courses, this allows the API to decide which courses to show, when necessary. We're going to want to support that here, could a separate PR but just want to bring that up here.

Reply:
Yeah could we do that in a separate ticket/PR, i'd like to avoid adding stuff to tickets once they've started 👍

Reply:
Works for me.

Reply:
https://trello.com/c/bqY8jqRL/3826-add-api-endpoint-api-public-v1-courses